Prevent header FOUC by forcing the navigation not to wrap, but rather be clipped

dd32 opened the pull request:
Fixes #41.

This forces the navigational menu to overflow horizontally, and be clipped. 

The simplest way to test this is to use a browser with JS disabled, you should be able to access the first few menu items.

Then with JS enabled, you should be able to access the overflow menu, and everything should be visually correct.

https://user-images.githubusercontent.com/767313/151741568-c358a21f-6c30-4fae-895c-95e0f0c8557c.mov

A `.has-overflow-menu` class was added to allow disabling the Overflow-X, as otherwise you'd have this:
(Note the background of the menu is clipped)
<img width="531" alt="Screen Shot 2022-01-31 at 3 05 52 pm" src="https://user-images.githubusercontent.com/767313/151741719-8468d9c3-af0f-441b-89de-1df1107a95a0.png">

The removal of the `left: 0` is to place the submenu correctly, I'm not sure why It's needed.. but I can't find any regressions and you otherwise end up with this, with 5ftf overlaying on Get Involved:
<img width="524" alt="Screen Shot 2022-01-31 at 3 08 19 pm" src="https://user-images.githubusercontent.com/767313/151741926-7f3da875-ddd8-4fd4-852a-d2ae3be79ccf.png">

dd32 committed:
Prevent header FOUC by forcing the navigation not to wrap, but rather be clipped.

ryelle committed:
Set flex alignment so that items wrap under the nav, not above

ryelle committed:
Fix nav menu item alignment

ryelle commented:
The `flex-wrap: no-wrap` was causing this for me:

<img width="492" alt="Screen Shot 2022-02-01 at 3 14 54 PM" src="https://user-images.githubusercontent.com/541093/152044144-354be248-1287-434d-86f1-0abd3d7cd8ee.png">

Instead, I switched the `align-items` on the whole header, so that the items would wrap under, rather than above the menu line. This made some alignment issues more obvious, so I've also fixed those.

dd32 commented:
> Instead, I switched the `align-items` on the whole header, so that the items would wrap under, rather than above the menu line. This made some alignment issues more obvious, so I've also fixed those.

Thanks, looks good to me!

I originally abandoned that approach as it was causing the items to wrap onto the next line (it still does) which would then be displayed below the header, and then quickly vanish. It looks like a later change caused the overflowed items to not be visible, resolving that approach.

ryelle commented:
>  It looks like a later change caused the overflowed items to not be visible, resolving that approach.

Yeah, I also had trouble with that - the `overflow: clip` + set height on the menu fixed that, though 👍🏻
